### DISCUSSION TITLE
Document Postgres Surfacer Table Structure

### DISCUSSION THREAD

**Discussion by theduke** [cloudprober/cloudprober#527] [Q&A] [answered]
Unless I'm missing it, there is no documentation for how the table must look like for the Postgres surfacer.

**Comment by manugarg** [comment, marked as answer]
We should definitely document it. For a quick reference though, postges surfacer supports two schemas:
* You dump all the labels in JSON format in a single column.
* You pre-define a label to column mapping through surfacer's `labelToColumn` field and create schema according to that.

Here is what the schema will look like for the first option:
```
CREATE TABLE metrics (
    time timestamp,
    metric_name varchar(80),
    value float8,
    labels jsonb
);
```

**Comment by manugarg** [comment]
Added some documentation to the config itself:
https://cloudprober.org/docs/config/surfacer/#cloudprober_surfacer_postgres_SurfacerConf
